File: server/laravel/c2c/app/Purchase.php
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Purchase extends Model
{
    //
    public function user(){
        return $this->belongsTo('App\User');
    }

    public function item(){
        return $this->belongsTo('App\Item');
    }

    public function notification(){
        return $this->morphMany('App\Notification', 'source');
    }
}

File: server/rails/c2c/app/models/purchase.rb
class Purchase < ApplicationRecord
  belongs_to :item
  belongs_to :user

  acts_as_paranoid

  after_create :notify_item_owner

  scope :for, ->(item) { where(item: item) }

  def notify_item_owner
    Notification.create! devices: item.owner.devices, title: I18n.t('message.item_purchased'), source: self
  rescue => e
    #TODO: handle notification error
  end
end
